draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -609,32 +588,69 @@ Elligator2 can be simplified with projective coordinates.
 
 ((TODO: write this variant))
 
-# Curve Transformations
-
-((TODO: write this section))
-
-# Cost Comparison
+## Cost Comparison
 
-The following table summarizes the cost of each hash_to_curve variant. We express this cost in 
+The following table summarizes the cost of each map2curve variant. We express this cost in 
 terms of additions (A), multiplications (M), squares (SQ), and square roots (SR). 
 
 ((TODO: finish this section))
 
 | Algorithm | Cost (Operations) | 
-| hash_to_curve_icart | TODO |
-| hash_to_curve_swu | TODO |
-| hash_to_curve_simple_swu | TODO |
-| hash_to_curve_elligator2 | TODO |
+| map2curve_icart | TODO |
+| map2curve_swu | TODO |
+| map2curve_simple_swu | TODO |
+| map2curve_elligator2 | TODO |
+
+# Random Oracles
+
+## Interface
+
+The generic interface for deterministic encoding functions to elliptic curves is as follows:
+
+~~~
+hash2curve(alpha)
+~~~
+
+where alpha is a message to encode on a curve. 
+
+## General Construction (FFSTV13)
+
+When applications need a Random Oracle (RO), they can be constructed from deterministic encoding 
+functions. In particular, let F : {0,1}^* -> E be a deterministic encoding function onto 
+curve E, and let H1 and H2 be two hash functions modeled as random oracles that map input 
+messages to the base field of E, i.e., Z_q. Farashahi et al. {{FFSTV13}} showed that the 
+following mapping is indistinguishable from a RO:
+
+~~~
+hash2curve(alpha) = F(H0(alpha)) + F(H1(alpha))
+~~~
+
+This construction works for the Icart, SWU, and Simplfied SWU encodings. 
+
+Here, H1 and H2 could be constructed as follows:
```

Comment:
H0 and H1

Reply:
>Section 9 (security recommendations) is a bit redundant now, since the whole piece is shaped around that comment.

Indeed. We can clean it up in a future PR.

> We should probably cite Brier et al. https://link.springer.com/chapter/10.1007/978-3-642-14623-7_13 as the primary reference for the general construction (they proved it for Icart function only though). And also can use the other generalized construction H(m) = f(h1(m)) + h2(m)G which you can show holds for Elligator

Do we really want to specify two generic mechanisms when one would suffice?

> Should double check the algorithm recommendations against table 1 in FFSTV13.

Yep, will do post-merge.